web/src/engine/predictive-text/worker-thread/src/main/correction/context-tracker.ts

```diff
@@ -31,6 +31,17 @@ function textToCharTransforms(text: string, transformId?: number) {
   return perCharTransforms;
 }
 
+/**
+ * Determines the proper 'last match' index for a tokenized sequence based on its edit path.
+ *
+ * In particular, this method is designed to handle the following case:
+ * ['to', 'apple', ' ', ''] => ['to', 'apply', ' ', 'n']
```

Comment:
It might be helpful to show the `editPath` this results in as well:

```suggestion
 * ['to', 'apple', ' ', ''] => ['to', 'apply', ' ', 'n']
 *  which results in the editPath ['match', 'subsitute',...
```